nucliadb/src/nucliadb/search/api/v1/hydrate.py

```diff
@@ -0,0 +1,758 @@
+
+                if config.related.neighbours.after is not None:
+                    hydrated.related.neighbours.after = []
+                    if config.related.neighbours.after > 0:
+                        for next_id in field_paragraphs.n_next(
+                            paragraph_id, config.related.neighbours.after
+                        ):
+                            await self._hydrate_related_paragraph(ParagraphId.from_string(next_id))
+                            hydrated.related.neighbours.after.append(next_id)
+
+            if config.related.parents:
+                hydrated.related.parents = []
+                for parent_id in field_paragraphs.parents(paragraph_id):
+                    await self._hydrate_related_paragraph(ParagraphId.from_string(parent_id))
+                    hydrated.related.parents.append(parent_id)
+
+            if config.related.siblings:
+                hydrated.related.siblings = []
+                for sibling_id in field_paragraphs.siblings(paragraph_id):
+                    await self._hydrate_related_paragraph(
+                        ParagraphId.from_string(sibling_id),
+                    )
+                    hydrated.related.siblings.append(sibling_id)
+
+            if config.related.replacements:
+                hydrated.related.replacements = []
+                for replacement_id in field_paragraphs.replacements(paragraph_id):
+                    await self._hydrate_related_paragraph(
+                        ParagraphId.from_string(replacement_id),
+                    )
+                    hydrated.related.replacements.append(replacement_id)
+
+        paragraph = field_paragraphs.get(paragraph_id)
+        if config.image:
+            if hydrated.image is None:
+                hydrated.image = hydration_models.HydratedParagraphImage()
+
+            if config.image.source_image:
+                source_image = paragraph.representation.reference_file
+                if (
+                    paragraph.kind
+                    in (
+                        resources_pb2.Paragraph.TypeParagraph.OCR,
+                        resources_pb2.Paragraph.TypeParagraph.INCEPTION,
+                    )
+                    and source_image
+                ):
+                    # Paragraphs extracted from an image store its original image
+                    # representation in the reference file. The path is incomplete
+                    # though, as it's stored in the `generated` folder
+                    image = await download_image(
+                        self.kbid,
+                        field_id,
+                        f"generated/{source_image}",
+                        # TODO: we assume all reference files are PNG images, but
+                        # this actually depends on learning so it's a dangerous
+                        # assumption. We should check it by ourselves
```

Comment:
I would say it's ok. just another implicit contract. I don't think we can do anything on our side. Leaving a comment is good enough at this point, imo !